custom_components/wellbeing/vacuum.py

```diff
@@ -0,0 +1,158 @@
+"""Vacuum platform for Wellbeing."""
+
+import asyncio
+import logging
+import math
+
+from homeassistant.components.vacuum import StateVacuumEntity, VacuumEntityFeature
+from homeassistant.components.vacuum import (
+    STATE_CLEANING,
+    STATE_PAUSED,
+    STATE_RETURNING,
+    STATE_IDLE,
+    STATE_DOCKED,
+    STATE_ERROR,
+)
+from homeassistant.const import Platform
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage,
+)
+
+from . import WellbeingDataUpdateCoordinator
+from .const import DOMAIN
+from .entity import WellbeingEntity
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+SUPPORTED_FEATURES = (
+    VacuumEntityFeature.START
+    | VacuumEntityFeature.STOP
+    | VacuumEntityFeature.PAUSE
+    | VacuumEntityFeature.RETURN_HOME
+    | VacuumEntityFeature.BATTERY
+)
+
+VACUUM_STATES = {
+    1: STATE_CLEANING,  # Regular Cleaning
+    2: STATE_PAUSED,
+    3: STATE_CLEANING,  # Stop cleaning
+    4: STATE_PAUSED,  # Pause Spot cleaning
+    5: STATE_RETURNING,
+    6: STATE_PAUSED,  # Paused returning
+    7: STATE_RETURNING,  # Returning for pitstop
+    8: STATE_PAUSED,  # Paused returning for pitstop
+    9: STATE_DOCKED,  # Charging
+    10: STATE_IDLE,
+    11: STATE_ERROR,
+    12: STATE_DOCKED,  # Pitstop
+    13: STATE_IDLE,  # Manual stearing
+    14: STATE_IDLE,  # Firmware upgrading
+}
+
+BATTERY_LEVELS = {
+    0: 0,
+    1: 10,
+    2: 30,
+    3: 50,
+    4: 70,
+    5: 90,
+    6: 100,
+}
```

Comment:
Make a property of this in Appliance

Reply:
Done. Copied the implementation of speed range and similarly used ranged_value_to_percentage, except that range value 1 has to be excluded to make the lower battery state 0%. Note that there was a bug in the prior implementation because the lowest value that the battery state can take is 1 and not 0, as I previously thought.